File: src/main/java/me/foodbag/hello/service/FoodService.java
package me.foodbag.hello.service;

import me.foodbag.hello.persistence.model.Food;

import java.util.List;

public interface FoodService {

  Food getFoodById(Long id);

  Food getFoodByName(String name);

  List<Food> getAllFoods();

  boolean exists(String name);

  Food save(Food food);
}

File: src/main/java/me/foodbag/hello/service/FoodService.kt
package me.foodbag.hello.service

import me.foodbag.hello.persistence.model.Food

interface FoodService {

    val allFoods: List<Food>

    fun getFoodById(id: Long?): Food

    fun getFoodByName(name: String): Food

    fun exists(name: String): Boolean

    fun save(food: Food): Food
}
